docs/source/url_configs.rst

```diff
@@ -0,0 +1,156 @@
+                tar.extractall(path=tmpdirname)
+                self.model = tf.keras.models.load_model(tmpdirname)
+                
+        def _get_model_file_name(self):
+            config_file = f'{self.algorithm}_model'
+            model_from_config = self.config.get(config_file, 'No file')
+            if model_from_config == 'No file':
+                raise ValueError(f'{__class__.__name__} should have {config_file} '
+                                f'provided as an option.')
+            if isinstance(model_from_config, str) and os.path.exists(model_from_config):
+                # Allow direct path specification
+                return model_from_config
+            if model_from_config is None:
+                # Allow None to be specified (disables processing for given posrec)
+                return model_from_config
+
+            # Use CMT
+            model_file = straxen.get_correction_from_cmt(self.run_id, model_from_config)
+            return model_file
+
+
+Notice how all the details on how to fetch the model file and convert it to a python object that is actually needed, is all hardcoded into the plugin. This is not desirable, the plugin should contain processing logic only.
+
+**How this could be refactored using `strax.URLConfig`:**
+
+.. code-block:: python
+
+    class PeakPositionsMLP(PeakPositionsBaseNT):
+        tf_model_mlp = straxen.URLConfig(
+            default=f'tf://'
+                    f'resource://'
+                    f'cmt://{algorithm}_model'
+                    f'?version=ONLINE'
+                    f'&run_id=plugin.run_id'
+                    f'&fmt=abs_path',
+        help='MLP model. Should be opened using the "tf" descriptor. '
+             'Set to "None" to skip computation',
+        cache=3,
+    )
+
+The details of where the model object is taken from can be determined by setting the model key of the context config
+The URL is the object being hashed, so it is important to only use pure URLs i.e the same URL should always refer to the same resource.
+
+The URL is evaluated recursively:
+  1) **tf://** - Loads a TF model from a path 
+  2) **resource://** - Loads a xenon resource by name (can also load web URLs), in this case returns a path to the file.
+  3) **cmt://** - Loads value from CMT, in this case it loads the name of the resource encoding the keras model.
+  4) **?version=ONLINE&run_id=plugin.run_id&fmt=abs_path** - Query is parsed and substituted (plugin.* are replaced with plugin attributes as evaluated at runtime) the values are then passed as keyword arguments to any protocols that include them in their signature. Everythin after the rightmost `?` character is considered the keyword arguments for the protocols.
+
+**Important** The URL arguments are sorted before they are passed to the plugin so that hashing is not sensitive to the order of the arguments.
+This is important to remember when performing tests.
```

Comment:
Sorted by?

Reply:
I am assuming the PR #827 will be merged at some point...